client/question/services/CurrentQuestionServiceSpec.js

```diff
@@ -90,3 +91,56 @@ describe('CurrentQuestionService', function() {
     });
   });
 });
+
+describe('CurrentQuestionService - Server', function() {
+  var CurrentQuestionService;
+  var $location;
+
+  var $httpBackend;
+  // In the Karma test environment, the deferred promise gets resolved only
+  // when $rootScope.$digest() is called.
+  var $rootScope;
+  beforeEach(module('tie'));
+  var setServerUrl = function(url) {
+    module('tieConfig', function($provide) {
+      $provide.constant('SERVER_URL', url);
+    });
+  };
+  var setDefaultQuestionId = function(id) {
+    module('tieConfig', function($provide) {
+      $provide.constant('DEFAULT_QUESTION_ID', id);
+    });
+  };
+
+  describe('behavior for invalid question - server', function() {
+
+    it('should return null', function(done) {
+      setServerUrl('http://katamari-dam.acy');
+      setDefaultQuestionId(null);
+
+      var PAGE_NOT_FOUND_ERROR = 404;
```

Comment:
I don't understand the point of having two separate constants here. They seem to be for post requests to the same URL. Why are they named differently?

Reply:
Added comments to explain. :)

Reply:
that's just my bad -- sorry, thought I changed that to a 400.